File: ExpressionToString.Tests/CompilerGenerated/Misc.cs
using System;
using Xunit;
using static ExpressionToString.Tests.Globals;
using static ExpressionToString.Tests.Runners;

namespace ExpressionToString.Tests {
    [Trait("Source", CSharpCompiler)]
    public class Misc {
        [Fact]
        public void Conditional() => RunTest(
            (int i) => i > 10 ? i : i + 10,
            "(int i) => i > 10 ? i : i + 10",
            "Function(i As Integer) If(i > 10, i, i + 10)"
        );

        [Fact]
        public void TypeCheck() {
            object o = "";
            RunTest(
                () => o is string,
                "() => o is string",
                "Function() TypeOf o Is String"
            );
        }

        [Fact]
        public void InvocationNoArguments() {
            Func<int> del = () => DateTime.Now.Day;
            RunTest(
                () => del(),
                "() => del()",
                "Function() del()"
            );
        }

        [Fact]
        public void InvocationOneArgument() {
            Func<int, int> del = (int i) => DateTime.Now.Day;
            RunTest(
                () => del(5),
                "() => del(5)",
                "Function() del(5)"
            );
        }
    }
}

File: Tests.Shared.VB/Misc.vb
Public Class Misc
    <Fact>
    Public Sub Conditional()
        RunTest(
            Function(i As Integer) If(i > 10, i, i + 10),
            "(int i) => i > 10 ? i : i + 10",
            "Function(i As Integer) If(i > 10, i, i + 10)"
        )
    End Sub

    <Fact>
    Public Sub ConstantNothingToObject()
        RunTest(
            Function() Nothing,
            "() => null",
            "Function() Nothing"
        )
    End Sub

    <Fact>
    Public Sub ConstantNothingToReferenceType()
        RunTest(Of String)(
            Function() Nothing,
            "() => null",
            "Function() Nothing"
        )
    End Sub

    <Fact>
    Public Sub ConstantNothingToValueType()
        RunTest(Of Integer)(
            Function() Nothing,
            "() => 0",
            "Function() 0"
        )
    End Sub

    <Fact>
    Public Sub TypeCheck()
        RunTest(
            Function() TypeOf "" Is String,
            "() => """" is string",
            "Function() TypeOf """" Is String"
        )
    End Sub

    <Fact>
    Public Sub InvocationNoArguments()
        Dim del = Function() Date.Now.Day
        RunTest(
            Function() del(),
            "() => del()",
            "Function() del()"
        )
    End Sub

    <Fact>
    Public Sub InvocationOneArgument()
        Dim del = Function(i As Integer) Date.Now.Day
        RunTest(
            Function() del(5),
            "() => del(5)",
            "Function() del(5)"
        )
    End Sub
End Class

